src/main/kotlin/com/sourcegraph/cody/listeners/CodyDocumentListener.kt

```diff
@@ -21,9 +21,9 @@ class CodyDocumentListener(val project: Project) : BulkAwareDocumentListener {
     val pastedCode = event.newFragment.toString()
     if (pastedCode.isNotBlank() && CodeEditorFactory.lastCopiedText == pastedCode) {
       CodeEditorFactory.lastCopiedText = null
-      ApplicationManager.getApplication().executeOnPooledThread {
-        GraphQlLogger.logCodeGenerationEvent(project, "keyDown:Paste", "clicked", pastedCode)
-      }
+      GraphQlLogger.logCodeGenerationEvent(project, "keyDown:Paste", "clicked", pastedCode)
+      TelemetryV2.sendCodeGenerationEvent(
+          project, feature = "keyDown", action = "paste", pastedCode)
```

Comment:
so we use both v1 and v2 at the moment, right?

Reply:
Yes, but we won't be adding new V1 events and existing ones will be removed in the near future.